service/history/handler.go

```diff
@@ -165,7 +177,7 @@ func (h *Handler) CreateEngine(
 func (h *Handler) Health(ctx context.Context) (*health.HealthStatus, error) {
 	h.startWG.Wait()
 	h.GetLogger().Debug("History health check endpoint reached.")
-	hs := &health.HealthStatus{Ok: true, Msg: common.StringPtr("history good")}
+	hs := &health.HealthStatus{Ok: true, Msg: common.StringPtr("OK")}
```

Comment:
nit: I'd consider setting the "Msg" to "Shutting down" when the host is in that state.

Reply:
Also, shouldn't we be reporting "unhealthy" status here? It seems like all endpoints are returning errShuttingDown even though the reported state is "healthy".

Reply:
Setting healthStatus to Ok=false has no effect for history - because we don't use rpc routing / load balancing mechanisms for forwarding requests to history. But I can still set it to ShuttingDown (even though it has no effect) - I didn't do it only because I didn't test that change in staging.

Reply:
Oh another reason I wasn't marking it unhealthy was because I wasn't rejecting calls to RespondDecision/ActivityTaskCompleted, RecordDecison/ActivityTaskStarted - because for these, most of the work is already done and rejecting it at last stage is probably not desirable. If shards have migrated, these calls will fail anyways - but I decided to accept them anyways till the end.

Reply:
Thanks for the info! That gives me another question though: why don't we continue accepting calls to all of the APIs as long as we still have the shard ownership, just like we do for Respond* and Record*?

Reply:
We could and I agree the benefit of this last step is questionable. But the reasoning is the following - for things like StartWF, SignalWF etc - we are about to start new work which can potentially take up to 200ms or more, when we know for sure that we are going down in another 500ms or so.. since we already waited ~5s or so, the shard should have most likely moved by now.. if not, we force the caller to go to the current shard owner.. everytime we reject, it triggers a backoff/retry on the caller (frontend) - where each retry will recompute the shard_owner.. so by rejecting, we are increasing the chance of request not failing and going to right owner.. 

Reason I say benefit is questionable is because - since we wait ~5s, its usually enough time for everyone on the cluster to learn that this instance is going down.. so its unlikely frontend will still forward calls to this node..

Reply:
OK, now I see that the time window is actually much shorter than I thought at first and with that context your thought process makes sense.